clang/lib/CodeGen/Targets/AMDGPU.cpp

```diff
@@ -52,6 +52,14 @@ class AMDGPUABIInfo final : public DefaultABIInfo {
   void computeInfo(CGFunctionInfo &FI) const override;
   RValue EmitVAArg(CodeGenFunction &CGF, Address VAListAddr, QualType Ty,
                    AggValueSlot Slot) const override;
+
+  llvm::FixedVectorType *
+  getOptimalVectorType(llvm::FixedVectorType *T,
+                       const LangOptions &Opt) const override {
+    if (T->getNumElements() == 3 && T->getScalarSizeInBits() == 32)
```

Comment:
Should use the datalayout to get the size

Reply:
Comment why. Also should add a fixme that we should check subtarget feature. Technically SI didn't have these